common/persistence/cassandra/cassandraPersistenceUtil.go

```diff
@@ -1817,9 +1825,40 @@ func createShardInfo(
 		info.ClusterReplicationLevel = make(map[string]int64)
 	}
 
+	if info.ClusterTransferProcessingQueueState == nil {
+		info.ClusterTransferProcessingQueueState = map[string]p.ProcessingQueueState{}
+		for cluster, ackLevel := range info.ClusterTransferAckLevel {
+			processingQueueState := createProcessingQueueState(ackLevel)
+			info.ClusterTransferProcessingQueueState[cluster] = processingQueueState
+		}
+	}
+
+	if info.ClusterTimerProcessingQueueState == nil {
+		info.ClusterTimerProcessingQueueState = map[string]p.ProcessingQueueState{}
+		for cluster, ackLevel := range info.ClusterTimerAckLevel {
+			processingQueueState := createProcessingQueueState(ackLevel.UnixNano())
+			info.ClusterTimerProcessingQueueState[cluster] = processingQueueState
+		}
+	}
```

Comment:
I think it's fine to let the application layer logic to handle this situation? If transfer/timer queue encounters an empty map on start, it knows this is a new shard, and can create a default queue.

My idea is that persistency layer shouldn't know anything related to default queue level or how queues are created. All it needs to do is persist and load.

Reply:
I was on the fence about this one. I agree that this isn't the perfect place to have this backfill logic but I favored consistency with similar backfill logic that was added to this layer earlier and wanted to avoid updating multiple places. After thinking about it again, I'm convinced that eliminating the backfill logic from this layer is the better option. I'll do it.